addons/website_slides/models/slide_slide.py

```diff
@@ -407,22 +407,25 @@ def _action_vote(self, upvote=True):
         slide_id = slide_partners.mapped('slide_id')
         new_slides = self_sudo - slide_id
         channel = slide_id.channel_id
+        karma_to_add = 0
 
         for slide_partner in slide_partners:
             if upvote:
```

Comment:
Maybe 

```
if upvote and slide_partner.vote != -1:
    new_vote = 1
elif not upvote and slide_partner.vote != 1:
    new_vote = -1
else:
    new_vote = 0
```

Would be easier to read?

Reply:
I added some code to protect karma addition against upvoting an already-upvoted or downvoted slide. As I added code in this if / else I would prefer to keep the current structure. I let you look at the final code.

Reply:
Just rewrite it. Maybe your structure will be better finally.

Do you love spam ?

Reply:
Nope, I keep the current one.

Reply:
What's in the box ?

Reply:
Gyproc ?

Reply:
Gyproc karma !

Reply:
https://www.youtube.com/watch?v=EuRKX-LrzDQ